toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -362,14 +363,14 @@ func customizeOSContents(ic *ImageCustomizerParameters) error {
 		ic.config.OS = &imagecustomizerapi.OS{}
 	}
 
-	// Check if the partition is using DM_verity_hash file system type.
-	// The presence of this type indicates that dm-verity has been enabled on the base image. If dm-verity is not enabled,
-	// the verity hash device should not be assigned this type. We do not support customization on verity enabled base
-	// images at this time because such modifications would compromise the integrity and security mechanisms enforced by dm-verity.
-	err := checkDmVerityEnabled(ic.rawImageFile)
+	// Check if dm-verity is enabled on the base image.
+	verityEnabled, err := isDmVerityEnabled(ic.rawImageFile)
 	if err != nil {
 		return err
 	}
+	if verityEnabled && len(ic.config.Storage.Verity) == 0 {
```

Comment:
I think this should be changed to: `verityEnabled  && !ic.customizeOSPartitions`

Currently, you can't specify verity without customizing partitions.

In the future, if the base image is verity-enabled and partitions aren't customized, then MIC should automatically re-apply the verity settings from the base image.

Also in the future, if the base image has the partitions to support verity but isn't enabled for verity, then we will add the ability to enable verity.